service/common/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -2014,6 +2025,12 @@ private Boolean getBooleanContextConfiguration(String configKey, boolean default
    * @return true if the exception is retryable
    */
   private static boolean isStorageProviderRetryableException(Exception ex) {
+    if (ex instanceof RuntimeException rex
+        && !RETRYABLE_CLOUD_EXCEPTION_CODES.contains(
+            IcebergExceptionMapper.extractHttpCodeFromCloudException(rex))) {
```

Comment:
We use `SdkException.retryable()` below, but why would we want to override what AWS SDK considers retryable here?

Reply:
That's a fair point. I changed it to use the retryable property for AWS/GCP. Azure doesn't have one though.